### DISCUSSION TITLE
What's the right way to wrap React Query?

### DISCUSSION THREAD

**Discussion by nandorojo** [TanStack/query#6072] [Q&A]
I want to build a library that has React Query as a dependency. I then want that library to set up its own query client, which is isolated to that lib. I assume this is fine, right? I figured the library could create its own query client, and rather than use React Query's context provider, it could use its own provider, and then pass the `client` to each hook. Does this sound right? Thanks!

**Comment by TkDodo** [comment]
> and then pass the client to each hook

yes, but you need v5 to do that, because that's where we added the functionality to pass a queryClient to the hooks as an argument.

**Reply by nandorojo** [reply]
cool, got it. we’re using v5 in this case.

if i’m using persistence too, should I just fork the persistence provider?

**Reply by TkDodo** [reply]
Yes
